packages/airnode-protocol/test/rrp/AirnodeRrpV0DryRun.sol.js

```diff
@@ -0,0 +1,176 @@
+      it('returns `true` and emits FulfilledRequest', async function () {
+        const endpointId = utils.generateRandomBytes32();
+        const requestTimeParameters = utils.generateRandomBytes();
+        const requestId = hre.ethers.utils.keccak256(
+          hre.ethers.utils.solidityPack(
+            [
+              'uint256',
+              'address',
+              'address',
+              'uint256',
+              'address',
+              'bytes32',
+              'address',
+              'address',
+              'address',
+              'bytes4',
+              'bytes',
+            ],
+            [
+              (await hre.ethers.provider.getNetwork()).chainId,
+              airnodeRrp.address,
+              rrpRequester.address,
+              (await airnodeRrp.requesterToRequestCountPlusOne(rrpRequester.address)).sub(1),
+              airnodeAddress,
+              endpointId,
+              roles.sponsor.address,
+              sponsorWalletAddress,
+              rrpRequester.address,
+              rrpRequester.interface.getSighash('fulfill'),
+              requestTimeParameters,
+            ]
+          )
+        );
+        const fulfillData = hre.ethers.utils.defaultAbiCoder.encode(['uint256', 'string'], ['123456', 'hello']);
+        const signature = await airnodeWallet.signMessage(
+          hre.ethers.utils.arrayify(
+            hre.ethers.utils.keccak256(hre.ethers.utils.solidityPack(['bytes32', 'bytes'], [requestId, fulfillData]))
+          )
+        );
+        const staticCallResult = await airnodeRrpDryRun.callStatic.fulfill(
+          requestId,
+          airnodeAddress,
+          rrpRequester.address,
+          rrpRequester.interface.getSighash('fulfill'),
+          fulfillData,
+          signature,
+          { gasLimit: 500000 }
+        );
+        expect(staticCallResult.callSuccess).to.equal(true);
+        expect(staticCallResult.callData).to.equal('0x');
+        await expect(
+          airnodeRrpDryRun.fulfill(
+            requestId,
+            airnodeAddress,
+            rrpRequester.address,
+            rrpRequester.interface.getSighash('fulfill'),
+            fulfillData,
+            signature,
+            { gasLimit: 500000 }
+          )
+        )
+          .to.emit(airnodeRrpDryRun, 'FulfilledRequest')
+          .withArgs(airnodeAddress, requestId, fulfillData);
+        const estimatedGas = await airnodeRrpDryRun.estimateGas.fulfill(
+          requestId,
+          airnodeAddress,
+          rrpRequester.address,
+          rrpRequester.interface.getSighash('fulfill'),
+          fulfillData,
+          signature
+        );
+        // The estimated gas will be ~40,000. This number can slightly change
+        // with future forks, and differ greatly between roll-ups and regular
+        // chains.
+        expect(estimatedGas.toNumber()).to.greaterThan(30000);
+        expect(estimatedGas.toNumber()).to.lessThan(50000);
```

Comment:
👍🏻 
Just being curious, what would the gas more or less be when calling `estimateGas` on `AirnodeRrpV0`? I wonder if there is any value in comparing the 2 values in another `expect()` just to be safe.

Reply:
Yes, I agree. I'm just trying to see if we can find a more future-proof way of testing the estimageGas() response by using a relative comparison instead of hardcoded numbers that may become obsolete on different chains, etc. But I guess we can do that once (and if) this becomes an issue.

Reply:
I was mainly testing that it returns some kind of a number without erroring here. I don't think the value is very important as long as it's something completely wrong like `-5` or `10000000000000000000`.